### PR TITLE
[DS-81] Stepper

### PR DESCRIPTION
**Pull request by allyssonsantos**
This PR implements Stepper component:

image

### CODE REVIEW

File: packages/yoga/src/Stepper/customPropType.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { number, checkPropTypes } from 'prop-types';
+
+const customPropType = (props, propName, componentName) => {
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/customPropType.js]:
```suggestion
const limitPropType = (props, propName, componentName) => {
```

File: packages/yoga/src/Stepper/native/Dots.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    font-weight: ${stepper.label.font.weight};
+    text-align: center;
+  `,
+);
+
+const Dots = ({ activeStep, labels }) => (
+  <Wrapper>
+    {labels.map((label, index) => (
+      <DotWrapper key={label}>
+        <Dot index={index} activeStep={activeStep} />
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/native/Dots.jsx]: What do you think about using `activeDot` here and change this prop to `active`?

File: packages/yoga/src/Stepper/native/Line.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    background-color: ${stepper.line.backgroundColor.active};
+    width: ${activeStep <= 0 ? 0 : (activeStep / totalSteps) * 100}%;
+    height: 4px;
+`,
+);
+
+const Line = ({ activeStep, totalSteps }) => (
+  <Wrapper>
+    <InactiveLine />
+    <ActiveLine activeStep={activeStep} totalSteps={totalSteps} />
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/native/Line.jsx]:
```suggestion
    <ActiveLine width={activeStep <= 0 ? 0 : (activeStep / totalSteps) * 100} />
```

File: packages/yoga/src/Stepper/native/Stepper.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      <Line
+        activeStep={activeStep}
+        totalSteps={React.Children.count(children) - 1}
+      />
+      <Dots
+        activeStep={activeStep}
+        labels={React.Children.map(children, child => child.props.label)}
+      />
+    </LineWrapper>
+    {children[activeStep]}
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/native/Stepper.jsx]:
```suggestion
    {React.Children.toArray(children)[activeStep]}
```

File: packages/yoga/src/Stepper/native/Stepper.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  </Wrapper>
+);
+
+Stepper.displayName = 'Stepper';
+
+Stepper.propTypes = {
+  /** Must be an Stepper.Step component. */
+  children: node,
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/native/Stepper.jsx]: Why not [this](https://gist.github.com/gaurav5430/c491b7aba253a2e9c17a57c028ea92a7#file-custom-proptypes-js)?

File: packages/yoga/src/Stepper/web/Dots.jsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  `,
+);
+
+const Label = styled.span(
+  ({
+    theme: {
+      components: { stepper },
+    },
+  }) => `
+    width: 95px;
```
Comment (ggdaltoso) [inline packages/yoga/src/Stepper/web/Dots.jsx]: 👍

File: packages/yoga/src/ThemeProvider/themes/BaseTheme.js [hunk 7]
```diff
@@ -23,6 +23,36 @@ const BaseTheme = ({ primary, secondary }) => {
...
+        font: {
+          size: fontSizes[1],
+          weight: fontWeights.bold,
+        },
+        color: {
+          active: colors.primary[3],
+          inactive: colors.gray[3],
+        },
+      },
+      content: {
```
Comment (ggdaltoso) [inline packages/yoga/src/ThemeProvider/themes/BaseTheme.js]: This is not required anymore. Right?

File: packages/yoga/src/Stepper/customPropType.js (hunk 1 above)
Comment (allyssonsantos) [reply packages/yoga/src/Stepper/customPropType.js]: Thanks!

File: packages/yoga/src/Stepper/native/Dots.jsx (hunk 2 above)
Comment (allyssonsantos) [reply packages/yoga/src/Stepper/native/Dots.jsx]: Great suggestion! <3

File: packages/yoga/src/Stepper/native/Line.jsx (hunk 3 above)
Comment (allyssonsantos) [reply packages/yoga/src/Stepper/native/Line.jsx]: Done

File: packages/yoga/src/Stepper/native/Stepper.jsx (hunk 4 above)
Comment (allyssonsantos) [reply packages/yoga/src/Stepper/native/Stepper.jsx]: Done!

File: packages/yoga/src/Stepper/native/Stepper.jsx (hunk 5 above)
Comment (allyssonsantos) [reply packages/yoga/src/Stepper/native/Stepper.jsx]: Done!

File: packages/yoga/src/ThemeProvider/themes/BaseTheme.js (hunk 7 above)
Comment (allyssonsantos) [reply packages/yoga/src/ThemeProvider/themes/BaseTheme.js]: Right!

File: packages/yoga/src/Stepper/web/Stepper.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+const Root = styled.div`
+  width: 100%;
+`;
+
+const LineWrapper = styled.div(
+  ({ theme: { spacing } }) => `
+  width: 100%;
+  height: 46px;
+
+  padding: 0 ${spacing.xxlarge}px;
```
Comment (victormath12) [inline packages/yoga/src/Stepper/web/Stepper.jsx]: this padding should use the component token, not the theme token right?

from:
```
 ${spacing.xxlarge}
```

to: 
```
${stepper.padding.left}
and
${stepper.padding.right}
```

**Pull request merged**